src/panels/ConfigPanel/ConfigPanel.tsx

```diff
@@ -268,110 +269,100 @@ export default function ConfigPanel(): JSX.Element {
     <Panel
       checkCanClose={checkCanClose}
       data-testid={TestId.PANEL}
+      drawer={
+        hasConfig ? (
+          <>
+            <Button onClick={handleCancel}>{Label.CANCEL_BUTTON}</Button>
+            <Button
+              appearance="positive"
+              className={classnames("config-panel__save-button", {
+                "is-active": savingConfig,
+              })}
+              onClick={handleSubmit}
+              disabled={!enableSave || savingConfig}
+            >
+              {!savingConfig ? (
+                Label.SAVE_BUTTON
+              ) : (
+                <>
+                  <Spinner isLight />
+                  <span>Saving&hellip;</span>
+                </>
+              )}
+            </Button>
+          </>
+        ) : null
+      }
       panelClassName="config-panel"
-      isSplit
+      isSplit={hasConfig}
       loading={isLoading}
       ref={scrollArea}
       title={
-        <h5>
+        <>
           {appName && charm ? (
             <CharmIcon name={appName} charmId={charm} />
           ) : null}{" "}
           {appName}
-        </h5>
+        </>
       }
       onRemovePanelQueryParams={handleRemovePanelQueryParams}
+      splitContent={
+        hasConfig ? (
+          <div className="config-panel__description">
+            {selectedConfig ? (
+              <FadeIn key={selectedConfig.name} isActive={true}>
+                <>
+                  <h4 className="p-heading--5">Configuration description</h4>
+                  <h5>{selectedConfig.name}</h5>
+                  <pre>{selectedConfig.description}</pre>
+                </>
+              </FadeIn>
+            ) : (
+              <div className="u-align--center">
+                <NoDescriptionMessage />
+              </div>
+            )}
+          </div>
+        ) : null
+      }
     >
       <>
-        {!isLoading && (!config || Object.keys(config).length === 0) ? (
```

Comment:
This one has a few changes. The split wrappers were moved to the panel component. The right hand side content was moved to the `splitContent` prop and the buttons to the drawer.